snuba/subscriptions/scheduler_processing_strategy.py

```diff
@@ -463,6 +463,13 @@ def submit(self, message: Message[CommittableTick]) -> None:
             )
             return
 
+        # Record the amount of time between the message timestamp and when scheduling
+        # for that timestamp occurs
+        self.__metrics.timing(
+            "scheduling_latency",
+            (time.time() - datetime.timestamp(message.timestamp)) * 1000,
+        )
```

Comment:
No, it doesn't include the jitter since the message timestamp here refers to the commit log message timestamp. This is equivalent to `receive_latency` that is currently tracked towards the start of the old subscriptions pipeline before a jitter is applied.

We already record executor latency. That one includes the jitter, and we can subtract the time shift from that one if we like. I think it's useful to record both scheduler and executor latency separately.

Reply:
> We already record executor latency. That one includes the jitter, and we can subtract the time shift from that one if we like. I think it's useful to record both scheduler and executor latency separately.

Makes sense. And yes I would remove the time shift as it makes the latency metrics not very representative.

Reply:
Though I am not sure we can just say the scheduler metric is equivalent to the old metric. There is another topic/consumer in between the moment where we log the query and the execution. This can introduce delay and backlog on its own.
So we should have a proper metric in the executor.

Reply:
It's the closest possible to the old metric we can do given that that happened at the start of the pipeline. But I don't think the old metric is the best metric for the new pipeline anymore. If we want to have a single metric only we'd probably be better off with something more end to end as our primary metric. Still I think it's useful to have both the scheduler and executor latency metric, which is what this PR was about.